### PR TITLE
Upload empty archive when all upload files were filtered out

### PR DESCRIPTION
**Pull request by sverdlov93**
- [ ] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: artifactory/services/upload.go [hunk 1]
```diff
@@ -350,6 +355,15 @@ func scanFilesByPattern(uploadParams UploadParams, rootPath string, progressMgr
 	return nil
 }
 
+func shouldUploadAnEmptyArchive(archive string, paths []string) bool {
+	if len(paths) == 0 &&
+		archive != "" &&
+		strings.ToLower(os.Getenv("JFROG_CLI_UPLOAD_EMPTY_ARCHIVE")) == "true" {
+		return true
+	}
+	return false
```
Comment (Or-Geva) [inline artifactory/services/upload.go:364]:
```suggestion
	return len(paths) == 0 &&
		archive != "" &&
		strings.ToLower(os.Getenv("JFROG_CLI_UPLOAD_EMPTY_ARCHIVE")) == "true" 
```

File: artifactory/services/upload.go [hunk 2]
```diff
@@ -321,6 +321,11 @@ func scanFilesByPattern(uploadParams UploadParams, rootPath string, progressMgr
 	// 'uploadedDirs' is in use only when we need to upload folders with flat=true.
 	// 'uploadedDirs' will contain only local directories paths that have been uploaded to Artifactory.
 	var uploadedDirs []string
+	if shouldUploadAnEmptyArchive(uploadParams.Archive, paths) {
```
Comment (Or-Geva) [inline artifactory/services/upload.go:324]: Let's move`uploadedDirs` to line 320 and add a log message that indicates JFROG_CLI_UPLOAD_EMPTY_ARCHIVE mode is detected

**Review by Or-Geva — CHANGES_REQUESTED** [review]
I think we need to add a test for `JFROG_CLI_UPLOAD_EMPTY_ARCHIVE` flow.

File: artifactory/services/upload.go [hunk 3]
```diff
@@ -350,6 +355,15 @@ func scanFilesByPattern(uploadParams UploadParams, rootPath string, progressMgr
 	return nil
 }
 
+func shouldUploadAnEmptyArchive(archive string, paths []string) bool {
+	if len(paths) == 0 &&
+		archive != "" &&
+		strings.ToLower(os.Getenv("JFROG_CLI_UPLOAD_EMPTY_ARCHIVE")) == "true" {
```
Comment (Or-Geva) [inline artifactory/services/upload.go:361]: Let's move `JFROG_CLI_UPLOAD_EMPTY_ARCHIVE` into const.

**Comment by sverdlov93** [comment]
> I think we need to add a test for `JFROG_CLI_UPLOAD_EMPTY_ARCHIVE` flow.
I added a test to JFrog CLI artifactory tests
@Or-Geva

**Pull request merged**